warehouse/query-core/src/main/java/datawave/query/jexl/nodes/ExceededOrThresholdMarkerJexlNode.java

```diff
@@ -131,24 +105,113 @@ public static JexlNode getExceededOrThresholdSource(JexlNode node) {
-                                new HashSet<>(Arrays.asList(StringUtils.split(String.valueOf(entry.getValue()), StringUtils.ESCAPE_CHAR, StringUtils.COMMA))));
-            } else if (entry.getKey().equalsIgnoreCase(FIELD_PROP)) {
-                returnParameters.put(FIELD_PROP, String.valueOf(entry.getValue()));
+        Object paramsObj = parameters.get(EXCEEDED_OR_PARAMS);
+        if (paramsObj != null)
+            return objectMapper.readValue(String.valueOf(paramsObj), ExceededOrParams.class);
+        else
+            return null;
+    }
+    
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static class ExceededOrParams {
+        private String field;
+        private String fstURI;
+        private Collection<String> values;
+        private Collection<String[]> ranges;
+        
+        private ExceededOrParams() {
+            
+        }
+        
+        public ExceededOrParams(String field, String fstURI) {
+            this.field = field;
+            this.fstURI = fstURI;
+        }
+        
+        public ExceededOrParams(String field, Collection<String> values) {
+            this(field, values, null);
+        }
+        
+        public ExceededOrParams(String field, Collection<String> values, Collection<Range> ranges) {
+            this.field = field;
+            init(values, ranges);
+        }
+        
+        private void init(Collection<String> values, Collection<Range> ranges) {
+            if (ranges == null || ranges.isEmpty()) {
+                this.values = values;
+            } else {
+                SortedSet<Range> rangeSet = new TreeSet<>();
+                
+                if (values != null)
+                    values.forEach(value -> rangeSet.add(new Range(new Key(value), new Key(value))));
+                
+                rangeSet.addAll(ranges);
+                
+                List<Range> mergedRanges = Range.mergeOverlapping(rangeSet);
+                Collections.sort(mergedRanges);
+                
+                this.ranges = new ArrayList<>();
+                for (Range mergedRange : mergedRanges) {
+                    if (mergedRange.getStartKey().getRow().equals(mergedRange.getEndKey().getRow())) {
+                        this.ranges.add(new String[] {String.valueOf(mergedRange.getStartKey().getRow())});
+                    } else {
+                        this.ranges.add(new String[] {(mergedRange.isStartKeyInclusive() ? "[" : "(") + String.valueOf(mergedRange.getStartKey().getRow()),
+                                String.valueOf(mergedRange.getEndKey().getRow()) + (mergedRange.isEndKeyInclusive() ? "]" : ")")});
+                    }
+                }
             }
         }
-        return returnParameters;
+        
+        private Range decodeRange(String[] range) {
+            if (range != null) {
+                if (range.length == 1) {
+                    return new Range(new Key(range[0]), new Key(range[0]));
```

Comment:
This could be `Range.exact(range[0])`

Reply:
Updated in the latest commit.

Reply:
Going to revert this change per the comment above.